### PR TITLE
(GH-1795) Add JSON schemas for Bolt configuration files

### LINKED ISSUE #1795: Add JSON schema for common yaml files.
We should have json schema's in the bolt repo for common yaml file formats in Bolt. These should live in a top level `/schemas` directory. 

- [x] inventory.yaml
- [x] bolt.yaml
- [x] project.yaml

### PR DESCRIPTION
**Pull request by beechtom**
This adds JSON schemas that can be used to validate `bolt.yaml`,
`inventory.yaml`, and `project.yaml` files.

Closes #1795 

!feature

* **Added JSON schemas for validating Bolt configuration files**
  ([#1795](https://github.com/puppetlabs/bolt/issues/1795))

  JSON schemas are now available for validating `bolt.yaml`,
  `inventory.yaml`, and `project.yaml` files.

### CODE REVIEW

File: schemas/bolt-config.schema.json [hunk 1]
```diff
@@ -0,0 +1,200 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Bolt Config",
+  "description": "Bolt Config bolt.yaml Schema",
+  "type": "object",
+  "additionalProperties": false,
```
Comment (adreyer) [inline schemas/bolt-config.schema.json:6]: we should only use this where bolt will blow up on unexpected keys.

Comment (beechtom) [reply schemas/bolt-config.schema.json:6]: Updated. Looks like the only place Bolt will blow up with unrecognized keys is `inventory.yaml` for groups, targets, and config.

File: schemas/bolt-inventory.schema.json [hunk 2]
```diff
@@ -0,0 +1,159 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Bolt Inventory",
+  "description": "Bolt Inventory inventory.yaml Schema",
+  "type": "object",
+  "additionalProperties": false,
+  "properties": {
```
Comment (nicklewis) [inline schemas/bolt-inventory.schema.json:7]: `version` should also be allowed here

File: schemas/bolt-inventory.schema.json [hunk 3]
```diff
@@ -22,6 +22,10 @@
     },
     "vars": {
       "$ref": "#/definitions/vars"
+    },
+    "version": {
+      "description": "The version of the inventory file.",
```
Comment (nicklewis) [inline schemas/bolt-inventory.schema.json:27]: This change shouldn't be in this commit. Nor should the one in `bolt-config.schema.json`

Comment (beechtom) [reply schemas/bolt-inventory.schema.json:27]: Whoops. Updated.

**Pull request merged**